skyrl-tx/tests/tinker/test_engine.py

```diff
@@ -254,3 +253,53 @@ def apply_step(request_id: int, model_id: str, request: types.OptimStepInput) ->
     # Expect a large gap in update magnitude between the two adapters.
     assert tiny_norm > 0
     assert default_norm / tiny_norm == pytest.approx(1e4, rel=5e-3)
+
+
+def test_gradient_checkpointing():
+    """
+    Verify gradient checkpointing doesn't affect loss values.
+    """
+    base_kwargs = dict(
+        base_model="Qwen/Qwen3-0.6B",
+        enforce_eager=False,
+        train_batch_size=2,
+        micro_batch_size=1,
+        max_lora_adapters=1,
+        max_lora_rank=4,
+    )
+
+    losses = []
+    for use_gradient_checkpointing in (False, True):
+        cfg = EngineConfig(**base_kwargs)
```

Comment:
I think it will be slightly cleaner / more typechecking to get rid of the dict and do
```python
cfg = EngineConfig(
    base_model="Qwen/Qwen3-0.6B",
    enforce_eager=False,
    train_batch_size=2,
    micro_batch_size=1,
    max_lora_adapters=1,
    max_lora_rank=4,
    gradient_checkpointing=use_gradient_checkpointing,
)
```